src/Console/ClearAvatarCache.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2024, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Console;
+
+use Friendica\App\BaseURL;
+use Friendica\Contact\Avatar;
+use Friendica\Core\L10n;
+use Friendica\Model\Contact;
+use Friendica\Model\Photo;
+use Friendica\Object\Image;
+use Friendica\Core\Config\Capability\IManageConfigValues;
+use Friendica\Core\Protocol;
+
+/**
+ * tool to clear the avatar file cache.
+ */
+class ClearAvatarCache extends \Asika\SimpleConsole\Console
+{
+	protected $helpOptions = ['h', 'help', '?'];
+
+	/**
+	 * @var $dba Friendica\Database\Database
+	 */
+	private $dba;
+
+	/**
+	 * @var $baseurl Friendica\App\BaseURL
+	 */
+	private $baseUrl;
+
+	/**
+	 * @var L10n
+	 */
+	private $l10n;
+
+	/**
+	 * @var IManageConfigValues
+	 */
+	private $config;
+
+	protected function getHelp()
+	{
+		$help = <<<HELP
+console clearavatarcache - Clear the file based avatar cache
+Synopsis
+	bin/console clearavatarcache
+
+Description
+	bin/console clearavatarcache
+		Clear the file based avatar cache
+
+Options
+	-h|--help|-? Show help information
+HELP;
+		return $help;
+	}
+
+	public function __construct(\Friendica\Database\Database $dba, BaseURL $baseUrl, L10n $l10n, IManageConfigValues $config, array $argv = null)
+	{
+		parent::__construct($argv);
+
+		$this->dba     = $dba;
+		$this->baseUrl = $baseUrl;
+		$this->l10n    = $l10n;
+		$this->config = $config;
+	}
+
+	protected function doExecute(): int
+	{
+		if ($this->config->get('system', 'avatar_cache')) {
+			$this->err($this->l10n->t('The avatar cache needs to be disabled to use this command.'));
```

Comment:
Is this something that can be done from the Administration screen or is it configuration file only?

Reply:
By now this can only be done via the file.

Reply:
If you’re going with a translated message, I’d prefer if we were extensive about what « disabling the avatar cache » actually entails.

Reply:
I added a hint to the local.config.php. Since this command does only make sense when you had activated this some time in the past, we don't need that much more information, I think.